app-creator/src/redux/reducer.ts

```diff
@@ -89,6 +92,43 @@ export const reducer: Reducer<AppCreatorStore, AppCreatorAction | AnyAction> = (
         ...state,
         eventType: action.payload.value ? EventType.reordering : EventType.none,
       };
+    case ADD_WIDGET_META_DATA:
+      return {
+        ...state,
+        template: {
+          ...state.template,
+          ...action.payload,
+        },
+      };
+    case UPDATE_WIDGET_META_DATA:
+      const updatedTemplate = { ...state.template };
+      updatedTemplate[action.payload.id][action.payload.attributeType][
+        action.payload.key
+      ] = action.payload.value;
+
+      const {
+        widgetRef,
+        style: { borderWidth, borderColor, borderType },
+      } = updatedTemplate[action.payload.id];
+
+      updatedTemplate[
+        action.payload.id
+      ].style.border = `${borderWidth} ${borderType} ${borderColor}`;
+
+      widgetRef.setStyle(updatedTemplate[action.payload.id].style);
+      updateUI(widgetRef, updatedTemplate);
+
+      return {
+        ...state,
+        template: updatedTemplate,
+      };
+    case REMOVE_WIDGET_META_DATA:
+      const newTemplate = { ...state.template };
+      delete newTemplate[action.payload.id];
+      return {
+        ...state,
+        template: newTemplate,
+      };
```

Comment:
It looks like this is deleting the widget entirely. If this is the case, could you rename `REMOVE_WIDGET_META_DATA` to something like `REMOVE_WIDGET` to reflect the real action?

Reply:
The reasoning behind it was that the store only stores the representation (meta data) that represents the widget. But I do agree it would make sense to change it to removeWidget.